Extension/lib/content-script/preload.js

```diff
@@ -130,6 +129,218 @@
+            }
+
+            styleElement.sheet.addRule(selectorText, cssText || "display: none!important");
+        };
+
+        /**
+         * Unhides elements which were previously hidden by the specified selector
+         */
+        var unhideBySelector = function(selectorText) {
+            var styleElement = document.getElementById(collapserStyleId);
+            if (!styleElement || !styleElement.sheet) {
+                return;
+            }
+            var iLength = styleElement.sheet.cssRules.length;
+            while (iLength--) {
+                var cssRule = styleElement.sheet.cssRules[iLength];
+                if (cssRule.selectorText == selectorText) {
+                    styleElement.sheet.deleteRule(iLength);
+                }
+            }
+
+            if (styleElement.sheet.cssRules.length === 0) {
+                // Schedule empty stylesheet removal
+                setTimeout(function() {
+                    // Double check stylesheet size
+                    if (styleElement.parentNode && styleElement.sheet && styleElement.sheet.cssRules.length === 0) {
+                        styleElement.parentNode.removeChild(styleElement);
+                    }
+                }, 100);
+            }
+        };
+
+        /**
+         * Searches for the specified elements in the hiddenElements collection
+         * and returns the first occurence.
+         */
+        var findHiddenElement = function(element) {
+            var iLength = hiddenElements.length;
+            while (iLength--) {
+                var hiddenElement = hiddenElements[iLength];
+                if (hiddenElement.node === element) {
+                    return hiddenElement;
+                }
+            }
+            return null;
+        };
+        
+        /**
+         * Hides specified element
+         */
+        var hideElement = function(element) {
+            var selectorText = getDomPath(element);
+            // First check if we have hidden it already
+            var hiddenElement = findHiddenElement(element);
+            if (hiddenElement && hiddenElement.selectorText === selectorText) {
+                // Nothing changed, we should do nothing
+                return;
+            }
+
+            var tagName = element.tagName.toLowerCase();
+            if (tagName === "frame" || tagName === "iframe") {
+                // Use specific style for frames due to these issues:
+                // https://github.com/AdguardTeam/AdguardBrowserExtension/issues/346
+                // https://github.com/AdguardTeam/AdguardBrowserExtension/issues/355
+                hideBySelector(selectorText, "visibility: hidden!important; height: 0px!important;");
+            } else {
+                hideBySelector(selectorText);
+            }
+
+            if (hiddenElement) {
+                // Remove redundant selector and save the new one
+                unhideBySelector(hiddenElement.selectorText);

```

Comment:
This will remove `selectorText { display: none; }` style.
We do it in case if hidden element selector has changed by the time of the second `hideElement` call (and this is a second call as hiddenElement already exists in the array).